featurizers/fingerprints.py

```diff
@@ -1,68 +1,234 @@
-import multiprocessing as mp
 from typing import Union
 
 import numpy as np
 import pandas as pd
 import rdkit.Chem as Chem
-from rdkit.Chem.rdMolDescriptors import GetMorganFingerprintAsBitVect
 
 from base import FingerprintTransformer
 
+"""
+If during multiprocessing occurs MaybeEncodingError, first check if there isn't thrown any exception inside
+worker function! (That error isn't very informative and this tip might save you a lot of time)
+"""
 
-class MorganFingerprintAsBitVect(FingerprintTransformer):
+"""
+fp_descriptors needs to be here or inside _transform() of a specific class (cannot be defined inside the __init__() of
+that class), otherwise pickle gets angry:
+        TypeError: cannot pickle 'Boost.Python.function' object
+"""
+
+
+class MorganFingerprint(FingerprintTransformer):
     def __init__(
         self,
         radius: int = 2,
         n_bits: int = 2048,
         use_chirality: bool = False,
         use_bond_types: bool = True,
         use_features: bool = False,
+        result_type: str = "default",
         n_jobs: int = 1,
     ):
+        assert result_type in ["default", "as_bit_vect", "hashed"]
```

Comment:
Po co "default"? Jako default zrób po prostu jedną z opcji. I w sumie czym się różnią hashed i bit vector?

Reply:
Z dokumentacji widzę:
- bit vectors : fpgen.GetFingerprint
- sparse (unfolded) bit vectors : fpgen.GetSparseFingerprint
- count vectors : fpgen.GetCountFingerprint
- sparse (unfolded) count vectors : fpgen.GetSparseCountFingerprint

Chodziło o bit i count fingerprinty?

Reply:
Z opisu używanych przez nas funkcji(są w rdMolDescriptors.py i w zależności od podanego argumentu używamy różnych z nich) wynika, że default to count, as_bit_vector to bit vector a hashed to sparse bit vector.

Używamy funkcji `GetMorganFingerprint` , `GetMorganFingerprintAsBitVect` i `GetHashedMorganFingerprint`

Czy ograniczyć te 3 funkcjonalności do tylko 2?

Reply:
A, bo mówiłem, żeby w osobnym PRze to zrobić. To generalnie już teraz przygotowałbym się, robiąc tu 4 dopuszczalne typy:
- count
- count_sparse
- bit
- bit_sparse

Póki co możemy obsługiwać tylko te które już są, a w następnym PRze przerobi się na interfejs "Generate"

Reply:
okej. To w takim razie wszystko jest ogarnięte. Przez weekend postaram się zrobić resztę

Reply:
To można robić merge. Przy okazji, obecna wersja libki zasadniczo nadawałaby się do instalacji, więc równolegle może iść kod i PR do instalacji pip'em.